File: renju/src/main/scala/renju/notation/Color.scala
package renju.notation

sealed trait Color {

  val flag: Flag

  val shift: Int

}

object Color {

  case object Black extends Color {

    val flag: Flag = new Flag(Flag.BLACK)

    val shift: Int = 0

  }

  case object White extends Color {

    val flag: Flag = new Flag(Flag.WHITE)

    val shift: Int = 1

  }

  def fromFlag(flag: Byte): Option[Color] = flag match {
    case Flag.BLACK => Some(Black)
    case Flag.WHITE => Some(White)
    case _ => Option.empty
  }

}

File: resrenju/src/notation/color.rs
#[derive(PartialEq, Eq, Clone, Copy)]
pub enum Color {
    Black,
    White
}

impl Color {

    pub fn reversed(&self) -> Color {
        match self {
            Color::Black => Color::White,
            Color::White => Color::Black
        }
    }

}
